addOns/websocket/src/main/java/org/zaproxy/zap/extension/websocket/ExtensionWebSocket.java

```diff
@@ -381,6 +406,40 @@ public void hook(ExtensionHook extensionHook) {
 
         eventPublisher = new WebSocketEventPublisher(this);
         this.addAllChannelSenderListener(eventPublisher);
+
+        // setup websocket passive scanner
+        if (extensionScript != null
+                && Control.getSingleton()
+                        .getExtensionLoader()
+                        .isExtensionEnabled(ExtensionAlert.NAME)) {
+
+            // Set up AlertManager so as to use it in WebSocketPassiveScannerManager
+            AlertManager alertManager =
+                    new AlertManager(
+                            Control.getSingleton()
+                                    .getExtensionLoader()
+                                    .getExtension(ExtensionAlert.class));
+
+            webSocketPassiveScannerManager = new WebSocketPassiveScannerManager(alertManager);
+
+            // Proxies which their's mode are equal to SERVER mode, they are ignored from passive
+            // scanner
+            webSocketPassiveScannerManager.setServerModeIgnored(true);
+            addAllChannelObserver(webSocketPassiveScannerManager.getWebSocketScannerObserver());
+
+            websocketPassiveScanScriptType =
+                    new ScriptType(
+                            SCRIPT_TYPE_WEBSOCKET_PASSIVE,
+                            "websocket.pscan.scripts.type.passive",
+                            getView() != null ? getScriptPassiveScanIcon() : null,
+                            true);
+            extensionScript.registerScriptType(websocketPassiveScanScriptType);
```

Comment:
We need to remove the script type on unload (we should also stop the passive scan).

Reply:
Fixed in 9718a29